arch/cpu/nrf/nrf-ieee-driver-arch.c

```diff
@@ -0,0 +1,1008 @@
+  }
+
+  LOG_DBG_("--->%u\n", nrf_radio_state_get(NRF_RADIO));
+
+  LOG_DBG("PACKETPTR=0x%08lx (rx_buf @ 0x%08lx)\n",
+          (uint32_t)nrf_radio_packetptr_get(NRF_RADIO), (uint32_t)&rx_buf);
+}
+/*---------------------------------------------------------------------------*/
+/* Retrieve an RSSI sample. The radio must be in RX mode */
+static int8_t
+rssi_read(void)
+{
+  uint8_t rssi_sample;
+
+  nrf_radio_task_trigger(NRF_RADIO, NRF_RADIO_TASK_RSSISTART);
+
+  while(nrf_radio_event_check(NRF_RADIO, NRF_RADIO_EVENT_RSSIEND) == false);
+  nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_RSSIEND);
+
+  rssi_sample = nrf_radio_rssi_sample_get(NRF_RADIO);
+
+  return -((int8_t)rssi_sample);
+}
+/*---------------------------------------------------------------------------*/
+/*
+ * Convert the hardware-reported LQI to 802.15.4 range using an 8-bit
+ * saturating multiplication by 4, as per the Product Spec.
+ */
+static uint8_t
+lqi_convert_to_802154_scale(uint8_t lqi_hw)
+{
+  return (uint8_t)lqi_hw > 63 ? 255 : lqi_hw *ED_RSSISCALE;
+}
+/*---------------------------------------------------------------------------*/
+/* Netstack API functions */
+/*---------------------------------------------------------------------------*/
+static int
+on(void)
+{
+  LOG_DBG("On\n");
+
+  if(radio_is_powered() == false) {
+    LOG_DBG("Not powered\n");
+    power_on_and_configure();
+  }
+
+  enter_rx();
+
+  ENERGEST_ON(ENERGEST_TYPE_LISTEN);
+  return NRF_COMMAND_OK;
+}
+/*---------------------------------------------------------------------------*/
+static int
+channel_clear(void)
+{
+  bool busy, idle;
+
+  LOG_DBG("channel_clear\n");
+
+  on();
+
+  /* Clear previous CCA-related events, if any */
+  nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_CCABUSY);
+  nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_CCAIDLE);
+  nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_CCASTOPPED);
+
+  LOG_DBG("channel_clear: CCACTRL=0x%08lx\n", NRF_RADIO->CCACTRL);
+
+  /* We are now in RX. Send CCASTART */
+  nrf_radio_task_trigger(NRF_RADIO, NRF_RADIO_TASK_CCASTART);
+
+  while((nrf_radio_event_check(NRF_RADIO, NRF_RADIO_EVENT_CCABUSY) == false) &&
+        (nrf_radio_event_check(NRF_RADIO, NRF_RADIO_EVENT_CCAIDLE) == false));
+
+  busy = nrf_radio_event_check(NRF_RADIO, NRF_RADIO_EVENT_CCABUSY);
+  idle = nrf_radio_event_check(NRF_RADIO, NRF_RADIO_EVENT_CCAIDLE);
+
+  LOG_DBG("channel_clear: I=%u, B=%u\n", idle, busy);
+
+  if(busy) {
+    return NRF_CCA_BUSY;
+  }
+
+  return NRF_CCA_CLEAR;
+}
+/*---------------------------------------------------------------------------*/
+static int
+init(void)
+{
+  LOG_DBG("Init\n");
+
+  last_rssi = 0;
+  last_lqi = 0;
+
+  timestamps.sfd = 0;
+  timestamps.framestart = 0;
+  timestamps.end = 0;
+  timestamps.mpdu_duration = 0;
+  timestamps.phr = 0;
+
+  /* Request the HF clock */
+  nrf_clock_event_clear(NRF_CLOCK, NRF_CLOCK_EVENT_HFCLKSTARTED);
+  nrf_clock_task_trigger(NRF_CLOCK, NRF_CLOCK_TASK_HFCLKSTART);
```

Comment:
You need to wait for the clock to start up - see the recent fix in #1478.
Could this be the source of your timer issues @Yagoor?